src/container.ts

```diff
@@ -491,6 +491,27 @@ export interface ContainerLifecycle {
 
 }
 
+/**
+ * RestartPolicy defines the restart behavior of individual containers in a pod.
+ * This field may only be set for init containers, and the only allowed value is "Always".
```

Comment:
How about adding a validation that normal containers don't specify this? We can do it here: 

https://github.com/cdk8s-team/cdk8s-plus/blob/0cd21c04e533e873c1698d173d395e4338e2a8ec/src/pod.ts#L192-L199